### ISSUE TITLE
jenkins deploy scala: could not find artifact ml.dmlc:xgboost4j_2.12

### ISSUE THREAD

**Issue by Le-Zheng** [intel-analytics/analytics-zoo#3228]
Deploy scala job failed when deploying Spark3: http://10.239.47.210:18888/job/ZOO-NB-Deploy-SCALA/937/console
It shows it can not download xgboost4j for scala 2.12
```
Downloading: https://oss.sonatype.org/content/groups/public/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/maven-metadata.xml


Downloading: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/xgboost4j_2.12-1.2.0-SNAPSHOT.pom

Downloading: https://oss.sonatype.org/content/groups/public/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/xgboost4j_2.12-1.2.0-SNAPSHOT.pom

[WARNING] The POM for ml.dmlc:xgboost4j_2.12:jar:1.2.0-SNAPSHOT is missing, no dependency information available
```
However, running on local shows good:
```
Downloading from sonatype: https://oss.sonatype.org/content/groups/public/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from XGBoost4J Snapshot Repo: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/ml/dmlc/xgboost4j_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from XGBoost4J Snapshot Repo: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/ml/dmlc/xgboost-jvm_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from sonatype: https://oss.sonatype.org/content/groups/public/ml/dmlc/xgboost-jvm_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from XGBoost4J Snapshot Repo: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/ml/dmlc/xgboost4j-spark_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from sonatype: https://oss.sonatype.org/content/groups/public/ml/dmlc/xgboost4j-spark_2.12/1.2.0-SNAPSHOT/maven-metadata.xml
Downloading from XGBoost4J Snapshot Repo: https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/snapshot/com/intel/analytics/zoo/zoo-core-dist-all/0.9.0-SNAPSHOT/maven-metadata.xml
```

**Comment by glorysdj** [comment]
https://s3-us-west-2.amazonaws.com/xgboost-maven-repo/
There's no 1.2.0-SNAPSHOT in that snapshot repo now, it is already promoted to 1.2.0 release
BTW, why do we use snapshot version of XGBoost4J instead of released versions(1.2.0, 1.3.0)?

We need to fix this, and backport to 0.9 branch before the 0.9 release. @helenlly

**Comment by Le-Zheng** [comment]
http://10.239.47.210:18888/job/ZOO-NB-Deploy-SCALA/942/ build successfully with our Nexus mirror. @helenlly @glorysdj @dding3
